homeassistant/components/sensor/mqtt.py

```diff
@@ -74,15 +82,39 @@ def async_added_to_hass(self):
         @callback
         def message_received(topic, payload, qos):
             """A new MQTT message has been received."""
+            # auto-expire enabled?
+            if self._expire_after > 0:
+                # Reset old trigger
+                if self._expiration_trigger:
+                    self._expiration_trigger()
+                    self._expiration_trigger = None
+
+                # Set new trigger
+                expiration_at = (
+                    dt_util.utcnow() + timedelta(seconds=self._expire_after))
+
+                self._expiration_trigger = async_track_point_in_utc_time(
+                    self.hass,
+                    self.value_is_expired,
+                    expiration_at)
+
             if self._template is not None:
-                payload = self._template.async_render_with_possible_json_value(
+                template = self._template
+                payload = template.async_render_with_possible_json_value(
```

Comment:
It looked to me that it was exactly 79 characters long. It should be fine. If it was in there it is OK for length.